unit_tests/test_aggregator.py

```diff
@@ -0,0 +1,415 @@
+    """
+    # mocker.patch("gsy_e_sdk.aggregator.AggregatorWebsocketMessageReceiver")
+    mocker.patch("gsy_e_sdk.aggregator.WebsocketThread", autospec=True)
+    mocker.patch("gsy_e_sdk.aggregator.ThreadPoolExecutor",
+                 autospec=True,
+                 return_value="test_set_callback_thread")
+
+
+@pytest.fixture(autouse=True)
+def mock_execute_batch_command_methods(mocker):
+    """
+    Fixture mocks methods and external functions and other methods used in
+    execute_batch_command.
+    """
+    mocker.patch("gsy_e_sdk.aggregator.ClientCommandBuffer.buffer_length",
+                 new_callable=PropertyMock, return_value=3)
+    mocker.patch("gsy_e_sdk.aggregator.ClientCommandBuffer.execute_batch",
+                 return_value=test_batch_command_dict)
+    mocker.patch("gsy_e_sdk.aggregator.Aggregator._post_request",
+                 return_value=["some_transaction_id", True])
+    mocker.patch("gsy_e_sdk.aggregator.AggregatorWebsocketMessageReceiver"
+                 + ".wait_for_command_response",
+                 return_value=test_response)
+    mocker.patch("gsy_e_sdk.aggregator.Aggregator."
+                 + "_all_uuids_in_selected_device_uuid_list",
+                 return_value=True)
+    mocker.patch("gsy_e_sdk.aggregator.log_bid_offer_confirmation")
+    mocker.patch("gsy_e_sdk.aggregator.log_deleted_bid_offer_confirmation")
+
+
+@pytest.fixture(name="aggregator_explicit")
+def fixture_aggregator_explicit():
+    """
+    Fixture creates and returns a aggregator with explicit parameters.
+    """
+    return Aggregator(aggregator_name=test_agg["name"],
+                      simulation_id=test_simulation["uuid"],
+                      domain_name=test_simulation["domain_name"],
+                      websockets_domain_name=test_simulation["websockets_domain_name"]
+                      )
+
+
+@pytest.fixture(name="aggregator_from_env")
+def fixture_aggregator_from_env(mocker):
+    """
+    Fixture creates and returns an aggregator with implicit parameters.
+    """
+    mocker.patch("gsy_e_sdk.clients.rest_asset_client.simulation_id_from_env",
+                 return_value=test_simulation["uuid"])
+    mocker.patch("gsy_e_sdk.clients.rest_asset_client.domain_name_from_env",
+                 return_value=test_simulation["domain_name"])
+    mocker.patch("gsy_e_sdk.clients.rest_asset_client.websocket_domain_name_from_env",
+                 return_value=test_simulation["websockets_domain_name"])
+
+    return Aggregator(aggregator_name=test_agg["name"])
+
+
+#####################################
+# DEF TESTS FOR CLASSES CONSTRUCTOR #
+
+class TestConstructor:
+    """Includes methods which test constructor."""
+
+    @staticmethod
+    def test_constructor__simulation_id_explicit(aggregator_explicit):
+        # this attribute is inherited from rest_asset_client
+        agg = aggregator_explicit
+        assert agg.simulation_id == test_simulation["uuid"]
```

Comment:
I think this line is not needed.
```suggestion
        assert aggregator_explicit.simulation_id == test_simulation["uuid"]
```